### DISCUSSION TITLE
Generating AST in antlr4 v4.10.1

### DISCUSSION THREAD

**Discussion by CHETOSRODRIGUEZ** [antlr/antlr4#3749] [Q&A]
Hi, my question is if i can to generate an AST in antlr4 WITH JAVA? 
i only found that i can to generate a CST 
                 TreeViewer viewr = new TreeViewer(rulesName,tree);
                  viewr.open();
but i wish an AST like the antlr2 version
IM USING VISITORS
Thank you!!!
![imagen_2022-06-13_224445973](https://user-images.githubusercontent.com/107454764/173494978-8a39c80c-dff1-4466-9cc4-5bf0825be4bb.png)

**Comment by alessiostalla** [comment]
ANTLR4 doesn't directly generate an AST. However, you can use listeners or visitors to build an AST for you. If you don't want to start from scratch, you could use a library like [Kolasu](https://github.com/Strumenta/kolasu) that provides the building blocks to develop an AST.
